### ISSUE TITLE
react native 0.59.10 androidx SOLVED

### ISSUE THREAD

**Issue by fabriziobertoglio1987** [ifsnow/react-native-infinite-flatlist-patch#2]
Thanks a lot for the awesome package.
This problem was solved and may be useful for troubleshooting

```
error: package androidx.core.content does not exist
```

I included in my `package.json` the postinstall code as by your instructions and I already had `"postinstall": "jetifier -r"` previously.

I ended up solving the issue as explained in [this post](https://github.com/react-native-community/react-native-webview/issues/580#issuecomment-521088110)

`yarn add jetifier`

and

```
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "test": "jest",
    "lint": "eslint *.js **/*.js",
    "postinstall": "jetifier -r; yarn run infinite-flatlist-patch",
    "postuninstall": "yarn run infinite-flatlist-patch"
  }
```

now the project installs witout problems
The package works perfectly!
IT IS SO AWESOME THANKS

**Comment by ifsnow** [comment]
@fabriziobertoglio1987 Thank you for telling me that my work has solved your problem.

**Comment by karthikshetty03** [comment]
Please explain how to do the same using npm

**Comment by ifsnow** [comment]
@karthikshetty03  The usage of `npm` is almost the same as `yarn`.

```bash
$ npm install jetifier --save-dev
$ npm run jetifier
```

**Comment by 18bhavna** [comment]
@fabriziobertoglio1987  I tried adding `"postinstall": "jetifier -r"` in packge.json, removed the node_modules and then re-installed them. But on running, my other libraries starts giving the error.

**Comment by ifsnow** [comment]
You should use `jetify`, not `jetifier`.

```
"postinstall": "jetify"
```

jetifier repo : https://github.com/mikehardy/jetifier

**Comment by 18bhavna** [comment]
Doing same, results in giving me error in another library  as
```
\node_modules\@react-native-community\netinfo\android\src\main\java\com\reactnativecommunity\netinfo\ConnectivityReceiver.java:14: error: package androidx.core.net does not exist
import androidx.core.net.ConnectivityManagerCompat;
```

**Comment by ifsnow** [comment]
@18bhavna Are you sure that postinstall is executed? You can run this script from a command line.

```
$ yarn jetify
```

**Comment by 18bhavna** [comment]
Yes it is working, but the their are patches installed for the library, which are not running.

**Comment by 18bhavna** [comment]
@ifsnow It is working. But for some the libraries it was generating error. while installing the patch.
Fixed them, by making the patch for the dependency,

**Comment by ifsnow** [comment]
@18bhavna Was my patch a problem? Can you tell me the fix you made for the solution? or Can you show me your `package.json?` I think it will help me.

**Comment by fabriziobertoglio1987** [comment]
> react-native-infinite-flatlist-patch@1.0.3 postinstall /Users/fabriziobertoglio/Documents/sourcecode/ReactNative/SurfApp/node_modules/react-native-infinite-flatlist-patch
> yarn run patch

sh: yarn: command not found
npm WARN acorn-jsx@5.1.0 requires a peer of acorn@^6.0.0 || ^7.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN react-addons-test-utils@15.6.2 requires a peer of react-dom@^15.4.2 but none is installed. You must install peer dependencies yourself.
npm WARN react-native@0.61.5 requires a peer of react@16.9.0 but none is installed. You must install peer dependencies yourself.
npm WARN react-native-webview@5.12.1 requires a peer of react-native@>=0.57 <0.60 but none is installed. You must install peer dependencies yourself.

npm ERR! code ELIFECYCLE
```
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! react-native-infinite-flatlist-patch@1.0.3 postinstall: `yarn run patch`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the react-native-infinite-flatlist-patch@1.0.3 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/fabriziobertoglio/.npm/_logs/2019-12-27T16_11_00_746Z-debug.log
```

**Comment by ifsnow** [comment]
@fabriziobertoglio1987 Thanks for your reporting. I didn't think of an environment without yarn. so, I removed yarn dependency of `postinstall` and published v1.0.4 right now.
